docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,164 @@
+
+        val exception = assertThrows<PRuntimeException> {
+            try {
+                val parseResult = parser.parse(query)
+                val plan = planner.plan(parseResult.statements[0], session).plan
+                val result = compiler.prepare(plan, Mode.STRICT())
+            } catch (e: PRuntimeException) {
+                // The exception should have been thrown from PartiQL components. You can add your handling here.
+                // Rethrow for validating the exception in test
+                throw e
+            }
+        }
+
+        assertEquals(exception.error.code(), PError.UNEXPECTED_TOKEN)
+        assertEquals(PErrorKind.SYNTAX, exception.error.kind.code())
+        assertEquals(Severity.ERROR, exception.error.severity.code())
+        assertEquals(SourceLocation(1, 14, 1), exception.error.location)
+        // Inspect properties from exception
+        assertEquals("UNRECOGNIZED", exception.error.get("TOKEN_NAME", String::class.java))
+    }
+
+    /**
+     * The example shows we registered a custom error handling in the context of major PartiQL components {@link PartiQLParser},
+     * {@link PartiQLPlanner} and {@link PartiQLCompiler} which records the errors or warning instead of throwing for deferred error processing.
+     * You can choose to treat warning as errors, or inhibit certain warnings or even ignore warnings.
+     * It might be a good idea for you to consolidate your logic of converting PErrors to user-presented messages and use that here.
+     */
+    @Test
+    fun `PartiQLParser with custom error handler`() {
+        class CustomPErrorListener : PErrorListener {
+            val errorCollection: MutableList<PError> = mutableListOf()
+
+            override fun report(error: PError) {
+                when (error.severity.code()) {
+                    // You may choose to customize the error message and then halt the flow by throwing the error, or you can record the error and then throw when it reaches maximum count
+                    // In this case, we record only without throwing
+                    Severity.ERROR -> errorCollection.add(error)
+                    // You may choose to customize the error message, treat warning as an error and then halt the flow by throwing, or you can suppress all warnings.
+                    // In this case, we record only without throwing
+                    Severity.WARNING -> errorCollection.add(error)
+                    else -> error("This shouldn't have occurred.")
+                }
+            }
+        }
+
+        // Register Error Listener
+        val listener = CustomPErrorListener()
+        val context = Context.of(listener)
+
+        val compiler = PartiQLCompiler.standard()
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val session = Session.empty()
+
+        val query = """
+            CREATE TABLE mytable;
+            INSERT INTO 'mytable VALUES (2,3);
```

Comment:
Similar to the comment in v0.14.9, let's just use normal queries (e.g. select-from-where, top-level expressions) rather than DDL/DML. PLK's implementation of DDL/DML (especially to everything after parsing) is pretty sparse. The majority of the present customers are just doing queries, so it may be helpful to keep the examples using queries.